src/modules/interp_met/WindNinja.cpp

```diff
@@ -90,23 +92,75 @@ void WindNinja::init(mesh& domain)
     {
         if( itr.find("Ninja") != std::string::npos)
         {
-          ++N_windfield;
+            ++N_windfield;
         }
     }
     N_windfield /= 3; // _U, _V, and speedup maps
 
     LOG_DEBUG << "Found " << N_windfield << " windfields";
+    try
+    {
+        // see if we have a manually specified number
+        int user_nwindfield = cfg.get<int>("N_windfield");
+
+        LOG_WARNING << "Override auto-detected windfield count with user specified value = " << user_nwindfield;
+
+        // we may have more in the param file than we specify if we have multiple averaging radii. Probably not an error but warn the user this makes sense!
+        if( user_nwindfield < N_windfield)
+            LOG_WARNING << "User specified " << user_nwindfield << " wind fields, but there are " << N_windfield << " in the param file. Make sure this makes sense.";
+
+        if(user_nwindfield > N_windfield)
+        {
+            CHM_THROW_EXCEPTION(module_error, "More wind fields were requested than what exist!");
+        }
+
+        N_windfield = user_nwindfield;
+    }
+    catch(...)
+    {
+        // not user specified, is ok
+    }
+
     if (N_windfield == 0)
     {
         CHM_THROW_EXCEPTION(module_error,"Could not find any required wind ninja maps");
     }
 
+    // ensure a user-specified L_avg value makes sense
+    L_avg = -1;
```

Comment:
I am wondering what could happen if L_avg is not defined in the cfg file whereas the mesh contains only WN parameters with an explicit values for L_avg. In this case, I guess WN will crash since it cannot find the wind maps in the mesh.

Reply:
Ok this should be resolved in cd2e258664b1116e7958134667863f4435894e4a now
It attempts to auto detect L_avg and if there are multiple options without a specified L_avg, it bails. Further, after detecting L_avg or not, it verifies all needed params are available in `init()`